### PR TITLE
DOC Update release process contributor guide

### PR DESCRIPTION
**Pull request by hildeweerts**
A few updates for the release guide based on my experiences.

## Tests

- [ ] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [x] user guide added or updated
- [ ] API docs added or updated
- [ ] example notebook added or updated

### CODE REVIEW

File: docs/contributor_guide/release.rst [hunk 1]
```diff
@@ -30,6 +29,10 @@ done on a clone of `fairlearn/fairlearn` and not on a fork).
 
         :code:`git push -u origin release/v<x.y>.X`
 
+       You may need to temporarilly add an exception to the
```
Comment (hildeweerts) [inline docs/contributor_guide/release.rst:32]:
```suggestion
       You may need to temporarily add an exception to the 
```

writing is hard

File: docs/contributor_guide/release.rst [hunk 2]
```diff
@@ -30,6 +29,10 @@ done on a clone of `fairlearn/fairlearn` and not on a fork).
 
         :code:`git push -u origin release/v<x.y>.X`
 
+       You may need to temporarily add an exception to the
```
Comment (riedgar-ms) [inline docs/contributor_guide/release.rst:32]: Have we filed an issue to puzzle out what I did wrong in the branch protections?

File: docs/contributor_guide/release.rst [hunk 3]
```diff
@@ -48,22 +51,21 @@ done on a clone of `fairlearn/fairlearn` and not on a fork).
...
-   hit "publish".
+#. On `GitHub's release page <https://github.com/fairlearn/fairlearn/releases>`_,
+   draft a new release. Choose the new tag, title the release `v<x.y.z>`, 
+   and post the changes from the release file within `docs/user_guide/installation_and_version_guide` 
+   into the description, then hit "publish".
 
 #. On the `main` branch, create a PR to:
 
-    #. Update the version in `__init__.py` to `x.y.z+1.dev0`
+    #. Update the version in `__init__.py` to `x.y+1.z.dev0`
```
Comment (riedgar-ms) [inline docs/contributor_guide/release.rst:61]: Good catch....

**Review by riedgar-ms — APPROVED** [review]

File: docs/contributor_guide/release.rst (hunk 2 above)
Comment (hildeweerts) [reply docs/contributor_guide/release.rst:32]: I'm not sure it's really "wrong" - heavily protecting these branches seems like a good idea to me.

**Review by romanlutz — APPROVED** [review]

**Pull request merged**